changes for module 2

rbaker opened the pull request:
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

JackMead commented on todo_app/data/session_items.py:
```diff
@@ -1,22 +1,41 @@
+    """
+    params = {
+        'key': os.environ['ACCOUNT_KEY'],
+        'token': os.environ['SECRET_KEY']
```
Here it looks like you're overloading the secret key as both the encryption key for the session (which is what SECRET_KEY was doing previously) and now using it to store the Trello token - keeping those two separate would make more sense

rbaker replied:
this has been done - i didnt realize that the `SECRET_KEY` was reserved. These have been renamed to `TRELLO_ACCOUNT_KEY` and `TRELLO_SECRET_KEY`

JackMead commented on todo_app/data/session_items.py:
```diff
@@ -1,22 +1,41 @@
+        'key': os.environ['ACCOUNT_KEY'],
+        'token': os.environ['SECRET_KEY']
+    }
+    response = requests.request('GET', 'https://api.trello.com/1/boards/607f1ee7c99b117ea947851a/lists', params=params)
```
Your board id isn't sensitive data, but it is something we'll to want to configure in different environments, so this (and all of the list ids) would be good candidates for also pulling out into environment variables

rbaker replied:
done

JackMead commented on todo_app/templates/index.html:
```diff
@@ -10,18 +10,36 @@ <h1 class="display-4">To-Do App</h1>
-        <a onclick="deleteItem({{ item.id }})" href="#">Delete</a>
+          <li>{{ item.name }} -- {{ item.desc }} -- {{ lists[item.idList]['name'] }}</li>
+          
+          {% if item.idList == '607f1ee7c99b117ea947851b' %}
+            <form id="completeItem{{item.id}}" method="POST" action="/complete/{{item.id}}/607f1ee7c99b117ea947851c">
```
It's quite hard to parse this logic - rather than matching list ids exactly, it might make more sense to:
* Split your `complete` endpoint into several, e.g. start, complete & restart
  * This would help make the urls much easier to reason about
  * It would also mean that the list ids don't have to be passed through to the template explicitly so we could then:
* Store the list names (or similar) on the Item objects rather than the list ids - so e.g. here we could compare like:
`if item.list_name == 'To Do'`

rbaker replied:
done

JackMead commented on todo_app/data/session_items.py:
```diff
@@ -1,22 +1,41 @@
     """
-    return sorted(session.get('items', _DEFAULT_ITEMS), key = custom_sort, reverse=True)
-
+    #return sorted(session.get('items', _DEFAULT_ITEMS), key = custom_sort, reverse=True)
```
If we're not using this any more then let's delete it rather than just commenting it out, we can always recover it with git (and by extension, the same for the custom_sort function)

rbaker replied:
done

JackMead reviewed (changes requested):
This looks good, I was able to run your app and view/create/update my todos stored in Trello. I've left a few thoughts below on some things we could improve, but in particular we should add some documentation - e.g. the requirements for a developer to have a Trello board set up, and e.g what environment variables need to be set. It would be nice to also keep your .env.template file up to date with the relevant variable names

rbaker commented:
> 
> 
> This looks good, I was able to run your app and view/create/update my todos stored in Trello. I've left a few thoughts below on some things we could improve, but in particular we should add some documentation - e.g. the requirements for a developer to have a Trello board set up, and e.g what environment variables need to be set. It would be nice to also keep your .env.template file up to date with the relevant variable names

I have addressed all comments and documentation has been added to a new section in the `README.md` file.
Please let me know if it is good to merge.

JackMead commented on .env.template:
```diff
@@ -4,3 +4,12 @@ FLASK_ENV=development
 # Change the following values for local development.
 SECRET_KEY=secret-key
+
+TRELLO_ACCOUNT_KEY=tello account key
```
Really minor, but typo in trello!

JackMead reviewed (approved):
Thanks for making those changes Rich, happy for you to merge this now
